doc/nrf/ug_matter_device_security.rst

```diff
@@ -0,0 +1,17 @@
+.. _ug_matter_device_security:
+
+Securing devices
+################
+
+.. contents::
+   :local:
+   :depth: 2
+
+.. include:: app_approtect.rst
+   :start-after: app_approtect_info_start
+   :end-before: app_approtect_info_end
+
+See :ref:`app_approtect` for more information.
+
+If you are working on a Matter over Thread end product based on the nRF52840 or the nRF5340 SoC, make sure to disable this mechanism to debug the firmware.
+
```

Comment:
Could we add one more point, that apart of enabling AP-Protect we suggest to disable debug serial port (e.g. UART) so that logs and shell commands are not accessible.

In general maybe it's worth to mention at the top, that below suggestions are for devices that are in production (not in development phase).